File: src/test/java/org/meetkt/catalogue/exception/NoProductFoundForBarcodeExceptionUnitTest.java
package org.meetkt.catalogue.exception;

import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

class NoProductFoundForBarcodeExceptionUnitTest {

    @Test
    void shouldReturnAnExceptionMessage() {
        NoProductFoundForBarcodeException exception = new NoProductFoundForBarcodeException("test-barcode");
        assertThat(exception.getMessage()).isEqualTo("No product found for test-barcode");
    }
}

File: src/test/kotlin/in/meetkt/catalogue/exception/NoProductFoundForBarcodeExceptionUnitTest.kt
package `in`.meetkt.catalogue.exception

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class NoProductFoundForBarcodeExceptionUnitTest {

    @Test
    fun shouldReturnAnExceptionMessage() {
        val exception = NoProductFoundForBarcodeException("test-barcode")

        assertThat(exception.message).isEqualTo("No product found for test-barcode")
    }
}